llvm/lib/Target/AMDGPU/AMDGPUAsmPrinter.cpp

```diff
@@ -990,7 +990,10 @@ void AMDGPUAsmPrinter::getSIProgramInfo(SIProgramInfo &ProgInfo,
   // dispatch registers are function args.
   unsigned WaveDispatchNumSGPR = 0, WaveDispatchNumVGPR = 0;
 
-  if (isShader(F.getCallingConv())) {
+  // Shaders that use the init.whole.wave intrinsic sometimes have VGPR
+  // arguments that are only added for the purpose of preserving their inactive
+  // lanes. Skip including them in the VGPR count.
+  if (isShader(F.getCallingConv()) && !MFI->hasInitWholeWave()) {
```

Comment:
Why do we need to make this conditional on `hasInitWholeWave`?

I think ignoring implicit VGPRs should either apply to all shaders (i.e. hw entrypoints) and functions or – if it causes problems for shaders – apply to all (gpu-callable) functions, but not shaders/hw entrypoints.

Reply:
I think it might cause problems for entry functions when using wave dispatch (I got this from an commit message, I don't really know how that works). But good point that I can probably make it apply to all non-entry functions.